### PR TITLE
Windows build

### PR DESCRIPTION
**Pull request by isaacbrodsky**
Addresses #32

This PR is based on @dfellis' branch working on Windows support and includes his commits too.

Adds ability to build on Windows, which was tested on Appveyor. We'll need to enable Appveyor on this repository, too.

There are a number of assumptions built into the build script which probably need to be documented. Specifically, if you want a 64-bit build, you must specify the environment variable `PYTHON_ARCH=64`. I'm not sure what the more idiomatic way of doing this is, perhaps we should check the processor architecture env variable?

Several issues were fixed to get it working on Windows:
* The type of `H3Index` should be `c_ulonglong`. On Windows, `long` and `c_long` are always 32 bits. I assigned this its own name in h3.py to be more explicit about what we mean.
* Fixed a test that was comparing floats for equality.
* Python 2.7: Fixed a test that checked for a number being an `int`. I presume it was a `long` instead.
* Python 2.7: Looks to me like formatting of indexes as strings was not working in Python 2.7 because it added a 'L' suffix.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -54,7 +54,9 @@ def __init__(self, attrs=None):
     },
     package_data={
         'h-py':
-        ['out/*.dylib' if platform.system() == 'Darwin' else 'out/*.so.*']
+        ['out/*.dylib' if platform.system() == 'Darwin' else (
+            'out/*.dll' if platform.system() == 'Windows' else
```
Comment (dfellis) [inline setup.py]: Now that I'm looking at this as a reviewer, I wonder if this should be `'out\\*.dll'` here to match how `libh3_path` is defined in `h3/h3.py` above?

Comment (isaacbrodsky) [reply setup.py]: Updated, but I standardized on forward slashes in `h3/h3.py` instead.

Comment (dfellis) [reply setup.py]: Does that work? I had the backslashes before because it wasn't concatenating the path correctly in Appveyor.

Comment (isaacbrodsky) [reply setup.py]: Appears to be working in Appveyor: https://ci.appveyor.com/project/isaacbrodsky/h3-py/builds/23110297

**Comment by karentycoon** [comment]
I've tried this patch in my windows machine, but got following errors:
`--Selecting Windows SDK version to target Windows 10.0.17134.
Cmake Error at
AMakeLists.txt:54 (project): Failed to run MSBuild command:
MSBuild.exe
to get the value of VCTargetsPath`
I've installed Visual Studio and powershelgl. Any steps I'm missing here?

**Comment by isaacbrodsky** [comment]
Which versions of Visual Studio, Windows, Python, and which architecture? I suspect this has to do with the script assuming Visual Studio 2015.

File: .install.sh [hunk 2]
```diff
@@ -21,14 +19,29 @@ pushd h3c
...
+  make
+  ls -l lib/libh3*
+  cp lib/libh3* ../h3/out
+  if [ -e ../build ] && [ -d ../build ]; then
+      LIBNAME=`ls ../build/ | grep '^lib'`
+      mkdir -p ../build/$LIBNAME/h3/out
+      cp lib/libh3* ../build/$LIBNAME/h3/out
+  fi
+else
+  # Assumed to be Windows, default to x64
```
Comment (karentycoon) [inline .install.sh]: I'm bit worried maybe some users have installed make in their windows environment? Maybe we can use uname to be more safe here?

**Comment by karentycoon** [comment]
> Which versions of Visual Studio, Windows, Python, and which architecture? I suspect this has to do with the script assuming Visual Studio 2015.

**Comment by karentycoon** [comment]
> Which versions of Visual Studio, Windows, Python, and which architecture? I suspect this has to do with the script assuming Visual Studio 2015.

It's my environment's problem. I'll try again.

**Comment by isaacbrodsky** [comment]
There's also beginning of a Cython binding, which may replace this approach: https://github.com/isaacbrodsky/h3-py/tree/cython I haven't tested yet if the Cython branch builds on Windows, but I would expect that it does.

File: .install.sh (hunk 2 above)
Comment (isaacbrodsky) [reply .install.sh]: I'm not sure what `uname` to look for. Maybe anything starting with `MINGW`? On my system it includes the Windows version as part of the OS name.

Another option might be to change from specifying the generator to specifying the architecture with `-A`, in that case we might be able to use CMake to invoke the build system, regardless of which build system is used.

Comment (isaacbrodsky) [reply .install.sh]: @karentycoon Updated, it now uses the `OS` environment variable to detect Windows.

**Comment by andrewong11** [comment]
Hi, I am quite new with programming. may I know if h3 can now be properly installed on windows? I have tried pip install h3, but it seems that the version of h3 installed is the one without these windows updates. I've also tried cloning the github repository, but i still couldn't import the h3 package. Or could you describe how I should install h3 with vs2017? thank you very much

**Comment by isaacbrodsky** [comment]
@andrewong11 I think a new version of H3-py needs to be published before it will be installable on Windows.

**Pull request merged**